### PR TITLE
Split the massive FinishGen.cpp file

### PR DESCRIPTION
**Pull request by 12xx12**
I did some changes to the 'FinishGen.cpp' file and noticed that it became a bit bloated. I split it to some smaller chunks

### CODE REVIEW

**Comment by tigerw** [comment]
Visual Studio wants to indent the closing parenthesis of a multi-line `if` when pasting, I think that may have broken the style in a few places.

**Comment by 12xx12** [comment]
DIdn't use Visual Studio. I'm going thru just to be shure

File: src/Generating/Finishers/FinishGenClumpTopBlock.cpp [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+		// Doesn't place if the blocks can't be placed. Checked value also depends on a_IsDoubleTall
+		if (Top + 1 + (a_IsDoubleTall ? 1 : 0) >= cChunkDef::Height)
+		{
+			continue;
+		}
+
+		auto GroundBlockType = a_ChunkDesc.GetBlockType(x, Top, z);
+		if (
+			(GroundBlockType == E_BLOCK_GRASS) || (
+				(GroundBlockType == E_BLOCK_MYCELIUM) && ((a_BlockType == E_BLOCK_RED_MUSHROOM) || (a_BlockType == E_BLOCK_BROWN_MUSHROOM))
```
Comment (12xx12) [inline src/Generating/Finishers/FinishGenClumpTopBlock.cpp:89]: here

File: src/Generating/Finishers/FinishGenFluidSprings.cpp [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+		{
+			ASSERT(!"Unhandled world dimension");
+			break;
+		}
+	}  // switch (dimension)
+	AString HeightDistribution = a_IniFile.GetValueSet(SectionName, "HeightDistribution", DefaultHeightDistribution);
+	if (!m_HeightDistribution.SetDefString(HeightDistribution) || (m_HeightDistribution.GetSum() <= 0))
+	{
+		LOGWARNING("[%sSprings]: HeightDistribution is invalid, using the default of \"%s\".", (a_Fluid == E_BLOCK_WATER) ? "Water" : "Lava", DefaultHeightDistribution.c_str()
+		);
```
Comment (12xx12) [inline src/Generating/Finishers/FinishGenFluidSprings.cpp:56]: here

File: src/Generating/Finishers/FinishGenOres.cpp [hunk 3]
```diff
@@ -0,0 +1,511 @@
...
+}
+
+
+
+
+
+const cFinishGenOres::OreInfos & cFinishGenOres::DefaultOverworldOres(void)
+{
+	static OreInfos res
+		{
```
Comment (12xx12) [inline src/Generating/Finishers/FinishGenOres.cpp:35]: here

File: src/Generating/Finishers/FinishGenOres.cpp [hunk 4]
```diff
@@ -0,0 +1,511 @@
...
+}
+
+
+
+
+
+const cFinishGenOres::OreInfos & cFinishGenOres::DefaultNetherOres(void)
+{
+	static OreInfos res
+		{
```
Comment (12xx12) [inline src/Generating/Finishers/FinishGenOres.cpp:56]: here

**Comment by 12xx12** [comment]
and maybe wait for #5129 #4993 to merge stuff only once

**Comment by tigerw** [comment]
I'm inclined to hold off splitting for now, mainly because we lose a lot of (easily accessible) source line level Git history that helps understanding why some change was made.

**Pull request closed without merging**